pkg/portutil/portutil.go

```diff
@@ -73,21 +75,30 @@ func ParseFlagP(s string) ([]gocni.PortMapping, error) {
 	if containerPort == "" {
 		return nil, fmt.Errorf("no port specified: %s", splitBySlash[0])
 	}
-
+	needAutoAssignHostPort := false
 	if hostPort == "" {
-		return nil, fmt.Errorf("automatic host port assignment is not supported yet (FIXME)")
-	}
-
-	startHostPort, endHostPort, err := nat.ParsePortRange(hostPort)
-	if err != nil {
-		return nil, fmt.Errorf("invalid hostPort: %s", hostPort)
+		commandConfig, err := cmd.Flags().GetBool("auto-hostport")
+		if err != nil || !commandConfig {
+			return nil, fmt.Errorf("no host port specified: %s", splitBySlash[0])
+		}
+		needAutoAssignHostPort = true
 	}
+	var startHostPort uint64
+	var endHostPort uint64
 
 	startPort, endPort, err := nat.ParsePortRange(containerPort)
 	if err != nil {
 		return nil, fmt.Errorf("invalid containerPort: %s", containerPort)
 	}
-
+	if !needAutoAssignHostPort {
+		startHostPort, endHostPort, err = nat.ParsePortRange(hostPort)
+		if err != nil {
+			return nil, fmt.Errorf("invalid hostPort: %s", hostPort)
+		}
+	} else {
+		startHostPort, endHostPort = startPort, endPort
+		logrus.Warnf("auto-hostport is enabled, host port range is %d-%d", startHostPort, endHostPort)
```

Comment:
Docker assigns a random port

Reply:
And we need integration tests

Reply:
I think if we need the same logic with docker. The nerdctl need an extra config to record which port has been allocated. 

Is it acceptable?

Reply:
>  The nerdctl need an extra config to record which port has been allocated.

No, https://github.com/lima-vm/lima/blob/943c90b13e38be32777b8f25be17c2491bb1421f/pkg/hostagent/hostagent.go#L181-L201

Reply:
You can also scan /proc/net/{tcp,tcp6} to find a free port

Reply:
Actually, in this first version of my thought before I make this PR, I have already thought about get a free port number by using listen "localhost:0". But I think it is not a good solution, here's my reason.

1. In the port range circumstance, we can't get a continuous port if we use the 'localhost:0' method.
2. In the Docker, the action of allocating port is trying to allocate the first free port in a specific port range which is not high frequency used in most circumstances (In Docker, the allocating range is from 

PTAL https://github.com/moby/moby/blob/master/libnetwork/portallocator/portallocator.go#L135-L176 and https://github.com/moby/moby/blob/master/libnetwork/portallocator/portallocator_unix.go#L7-L12

Reply:
Better to scan /proc/net/* then?

Reply:
I'm not sure we should think about the compatibility of the Windows platform. Or we just need support auto port allocated for the Linux platform.

Reply:
ping @AkihiroSuda

Reply:
No need to cover windows now

Reply:
Got it.